Shopify/shopify-cli
# support resource_url and public_url

### WHY are these changes introduced?

Closes:
https://github.com/Shopify/shopify-cli-extensions/issues/151
https://github.com/Shopify/shopify-cli-extensions/issues/152

### WHAT is this pull request doing?

* Passes `resource_url` to extensions development server as `resource.url`
* Passes `tunnel_url` as `public_url` (same as before)
* Passes `store`
* Pulls resource_url logic into a task, as `ArgoServe` was getting really long.

### How to test your changes?

* within the shopify-cli repo, run rake extensions:install
* give yourself the required beta flag: shopify config feature extension_server_beta --enable
* `cd` into an extension project
* run `shopify extension serve`

you should see the following output:

<img width="1079" alt="Screen Shot 2021-11-02 at 12 12 28 PM" src="https://user-images.githubusercontent.com/989784/139942817-20c97bbb-cc38-4ae7-809d-29b6ce1eebc3.png">

### Update checklist

~- [ ] I've added a CHANGELOG entry for this PR (if the change is public-facing)~ not public facing yet
